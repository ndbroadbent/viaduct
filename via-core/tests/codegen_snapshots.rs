use std::path::Path;

use anyhow::Result;
use via_core::{codegen, parser};

#[test]
fn generates_expected_outputs_for_article_fixture() -> Result<()> {
    let fixture = Path::new("tests/fixtures/article.via");
    let resources = parser::parse_file(fixture)?;
    assert_eq!(resources.len(), 1);

    let generation = codegen::generate(&resources)?;

    let mut files = generation.files;
    files.sort_by(|a, b| a.relative_path.cmp(&b.relative_path));

    for file in files {
        let path_str = file.relative_path.to_string_lossy().replace('/', "__");
        let snapshot_name = format!("article__{}", path_str);
        insta::assert_snapshot!(snapshot_name, file.contents);
    }

    Ok(())
}
