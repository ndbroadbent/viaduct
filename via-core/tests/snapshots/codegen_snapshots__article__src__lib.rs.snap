---
source: tests/codegen_snapshots.rs
expression: file.contents
---
// @generated by via-core. DO NOT EDIT BY HAND.

pub mod controllers;
pub mod models;
